securedrop/crypto_util.py

```diff
@@ -174,8 +179,27 @@ def decrypt(secret, plain_text):
     'Goodbye, cruel world!'
     """
     hashed_codename = hash_codename(secret, salt=SCRYPT_GPG_PEPPER)
-    return gpg.decrypt(plain_text, passphrase=hashed_codename).data
 
+    return gpg.decrypt(ciphertext, passphrase=hashed_codename).data
+
+# Modified version of function from python-gnupg to make sure we are passing a
+# steam/file-like object to GPG. We choose to do this conversion here to ensure
+# that we control the encoding and conversion type. See
+# https://github.com/freedomofpress/securedrop/issues/1360#issuecomment-239593607
+# for more on why this was implemented here.
+def _make_binary_stream(thing):
+    """Encode **thing**, then make it stream/file-like.
+
+    :param thing: The thing to turn into a encoded stream.
+    :rtype: ``io.BytesIO``
+    :returns: The encoded **thing**, wrapped in an ``io.BytesIO``.
+    """
+    encoding = "utf_8"
+
+    if type(thing) is not str:
+        thing = thing.encode(encoding)
+
+    return BytesIO(thing)
```

Comment:
Why not use `python-gnupg`'s `_make_binary_stream` with an explicit `encoding` parameter? This function seems like an unnecessary duplication of logic from the python-gnupg function.

One could argue that it's bad practice to use "internal" functions from other libraries (prefixed with `_`). Meh. DRY is a good principle. The original code in `crypto_util.encrypt` was also good:

```
if isinstance(plaintext, unicode):
    plaintext = plaintext.encode('utf8')
```